src/Pact/Native/Ops.hs

```diff
@@ -112,22 +114,52 @@ divDef = defRNative "/" divide' coerceBinNum
 powDef :: NativeDef
 powDef = defRNative "^" pow coerceBinNum ["(^ 2 3)"] "Raise X to Y power."
   where
-    pow :: RNativeFun e
-    pow = binop (\a b -> liftDecF (**) a b)
-                (\a b -> assert (b >= 0) "Integral power must be >= 0" $ liftOp (^) a b)
+  pow :: RNativeFun e
+  pow = binop (\a b -> liftDecF (**) a b)
+              (\a b -> assert (b >= 0) "Integral power must be >= 0" $ liftOp (^) a b)
+
+legalLogArg :: Literal -> Bool
+legalLogArg = \case
+  LInteger i -> i > 0
+  LDecimal i -> i > 0
+  _ -> False
+
+litGt0 :: Literal -> Bool
+litGt0 = \case
+  LInteger i -> i >= 0
+  LDecimal i -> i >= 0
+  _ -> False
 
 logDef :: NativeDef
 logDef = defRNative "log" log' coerceBinNum ["(log 2 256)"] "Log of Y base X."
   where
-    log' :: RNativeFun e
-    log' = binop (\a b -> liftDecF logBase a b)
-                 (\a b -> liftIntF logBase a b)
+  log' :: RNativeFun e
+  log' fi as@[TLiteral base _,TLiteral v _] = do
+    whenExecutionFlagSet FlagDisableFQVars $
+      when (not (litGt0 base) || not (legalLogArg v)) $ evalError' fi "Illegal base or argument in log"
```

Comment:
The name `FlagDisableFQVars` makes guarding this behind a flag genuinely confusing when reviewing. I wouldn't expect that fork to contain arithmetic fixes that have nothing to do with the fqn stuff. Proposing a second flag to differentiate the audit work from the other mem/inline/fqn work here.